### PR TITLE
feat: introduce navigator.language

### PR DESCRIPTION
**Pull request by AnInternetTroll**
Link to the spec: https://html.spec.whatwg.org/multipage/system-state.html#dom-navigator-language-dev
[~~To access `navigator.language` you **must** pass `--allow-env` as it can lead to fingerprinting.~~](#issuecomment-934847231)

What this PR **doesn't** add is `languagechange` which quite frankly is outside of my skill level.

### CODE REVIEW

**Comment by AnInternetTroll** [comment]
The spec says the following
> If the user is using an anonymizing service, then the value "en-US" is suggested; if all users of the service use that same value, that reduces the possibility of distinguishing the users from each other.

I would argue that if `--allow-env` isn't passed then `navigator.language` should be `en-US`. 

Also an additional note, on unix this would be more or less the same as getting the `LANG` env variable, so maybe `--allow-env=LANG` should allow `navigator.language` to return the correct language. This should be discussed though as it is not how it works on windows.

File: runtime/js/99_main.js [hunk 1]
```diff
@@ -321,6 +321,16 @@ delete Object.prototype.__proto__;
         return numCpus;
       },
     },
+    language: {
+      get() {
+        return os.language()
```
Comment (crowlKats) [inline runtime/js/99_main.js:326]: add `webidl.assertBranded(this, Navigator);`

File: runtime/js/99_main.js [hunk 2]
```diff
@@ -321,6 +321,16 @@ delete Object.prototype.__proto__;
...
+      }
+    },
+    languages: {
+      get() {
+        return os.languages()
```
Comment (crowlKats) [inline runtime/js/99_main.js:331]: add `webidl.assertBranded(this, Navigator);`

**Review by crowlKats — CHANGES_REQUESTED** [review]
Missing WorkerNavigator typings and property definition in 99_main

File: runtime/js/99_main.js (hunk 1 above)
Comment (crowlKats) [reply runtime/js/99_main.js:327]: (and `configurable` and `enumerable`)

**Comment by lucacasonato** [comment]
Actually, this can be enabled by default (no permissions), because the default language can already be extracted via `Intl` global.

**Review by crowlKats — DISMISSED** [review]
`languages` should be frozen

**Comment by AnInternetTroll** [comment]
Not stale, work has been done in rusty_v8 https://github.com/denoland/rusty_v8/pull/953

**Comment by bartlomieju** [comment]
This is now done and should be ready to land. @lucacasonato @crowlKats please review

**Review by lucacasonato — APPROVED** [review]
LGTM

**Review by bartlomieju — APPROVED** [review]
LGTM

File: runtime/worker_bootstrap.rs [hunk 3]
```diff
@@ -47,6 +48,7 @@ impl Default for BootstrapOptions {
       enable_testing_features: Default::default(),
       debug_flag: Default::default(),
       ts_version: Default::default(),
+      locale: "en-EN".to_string(),
```
Comment (zbraniecki) [inline runtime/worker_bootstrap.rs:51]: this seems like an error - there's no region `EN`.

**Review by zbraniecki — COMMENTED** [review]
fly-by review.

File: runtime/worker_bootstrap.rs (hunk 3 above)
Comment (bartlomieju) [reply runtime/worker_bootstrap.rs:51]: Thanks! Silly mistake on my part, I'll fix it

**Pull request merged**